addons/mail/static/src/models/thread/thread.js

```diff
@@ -1192,6 +1258,20 @@ function factory(dependencies) {
             return [['replace', this.messages.filter(message => message.isNeedaction)]];
         }
 
+        /**
+         * @returns {integer|undefined}
+         * @private
+         */
+        _computeMessageBeforeNewMessageSeparatorId() {
+            if (!this.lastSeenByCurrentPartnerMessageId) {
+                return 0;
+            }
+            if (!this.lastTransientMessage) {
+                return this.lastSeenByCurrentPartnerMessageId;
+            }
+            return Math.max(this.lastTransientMessage.id, this.lastSeenByCurrentPartnerMessageId);
+        }
+
```

Comment:
As suggested at the start: preventive return doing `clear` if we don't have a channel, or if we viewed all messages.